mutiny-core/src/nostr/nwc.rs

```diff
@@ -27,11 +29,100 @@ pub struct SingleUseSpendingConditions {
     pub amount_sats: u64,
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct TrackedPayment {
+    /// Time in seconds since epoch
+    pub time: u64,
+    /// Amount in sats
+    pub amt: u64,
+    /// Payment hash
+    pub hash: String,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub enum BudgetPeriod {
+    Day,
+    Week,
+    Month,
+    Year,
+    Seconds(u64),
+}
```

Comment:
Would be good to have some docs here about how it works. For instance, from reading the code, these reset at each interval right? 

- Budget resets at midnight
- Budget resets on sunday
- Budget resets on Sept 1
- Budget reset on Jan 1

Seconds it's a bit different right? It checks the last X seconds?